storage/innobase/lock/lock0lock.cc

```diff
@@ -546,6 +548,17 @@ static void wsrep_assert_no_bf_bf_wait(const lock_t *lock, const trx_t *trx)
 		return;
 	}
 
+	if (type_mode)
+	{
+		ib::error() << " Requested lock "
+			    << ((type_mode & LOCK_TABLE) ? "on table " : " on record ")
+			    << ((type_mode & LOCK_WAIT) ? " WAIT " : " ")
+			    << ((type_mode & LOCK_GAP) ? " GAP " : " ")
+			    << ((type_mode & LOCK_REC_NOT_GAP) ? " RECORD " : " ")
+			    << ((type_mode & LOCK_INSERT_INTENTION) ? " INSERT INTENTION " : " ")
+			    << ((type_mode & LOCK_X) ? " LOCK_X " : " LOCK_S ");
+	}
```

Comment:
This introduces an excessive number of calls to `operator<<()` functions. Why not just simply write the following:
```c++
	sql_print_error("InnoDB: Requested lock %03x", type_mode);
```
If someone is interested in particular values, 4-digit hexadecimal values should be much easier to filter (for example with regular expressions) than any such partially "decoded" output.

What is the check for `type_mode == 0` good for? It would seem to be an almost valid value that would correspond to an intention-shared record lock. (To my knowledge, intention locks are only possible on tables, not index records.)

Reply:
I do not agree on hexadecimal values, I prefer clear strings. type_mode = 0 is case when actual requested lock type is not known but your suggestion LOCK_NONE is good.